src/api-client.js

```diff
@@ -0,0 +1,107 @@
+const core = require('@actions/core')
+const github = require('@actions/github')
+const hc = require('@actions/http-client')
+
+// All variables we need from the runtime are loaded here
+const getContext = require('./context')
+
+async function getSignedArtifactUrl({ runtimeToken, workflowRunId, artifactName }) {
+  const { runTimeUrl: RUNTIME_URL } = getContext()
+  const artifactExchangeUrl = `${RUNTIME_URL}_apis/pipelines/workflows/${workflowRunId}/artifacts?api-version=6.0-preview`
+
+  const httpClient = new hc.HttpClient()
+  let data = null
+
+  try {
+    core.info(`Artifact exchange URL: ${artifactExchangeUrl}`)
+    const response = await httpClient.getJson(artifactExchangeUrl, {
+      Authorization: `Bearer ${runtimeToken}`
+    })
+
+    data = response?.result
+    core.debug(JSON.stringify(data))
+  } catch (error) {
```

Comment:
According to the [actions/http-client](https://www.npmjs.com/package/@actions/http-client#user-content-errors), it doesn't raise exceptions on non-success HTTP statuses like axios or octokit does.

Reply:
Ooo, good catch! 😬

Reply:
Turns out that's only _partially_ true. 😅 

Using `getJson` instead of `get` actually adds the potential for rejecting with an `HttpClientError` but it isn't consistent enough for my taste. I'm going to switch to use `get` instead of `getJson`, which _will_ let us have full control over the error handling (since it does none 😜).